catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/ToIcebergTypeVisitor.java

```diff
@@ -52,6 +54,19 @@ public static <T> T visit(Type type, ToIcebergTypeVisitor<T> visitor) {
     } else if (type instanceof Types.ListType) {
       Types.ListType list = (Types.ListType) type;
       return visitor.array(list, visit(list.elementType(), visitor));
+    } else if (type instanceof Types.StructType) {
+      Types.StructType struct = (Types.StructType) type;
+      Types.StructType.Field[] fields = struct.fields();
+      List<IcebergColumn> columns =
+          Arrays.stream(fields)
+              .map(
+                  field -> {
+                    return ConvertUtil.fromGravitinoField(field, 0);
+                  })
+              .collect(Collectors.toList());
+      IcebergTable mockTable =
+          new IcebergTable.Builder().withColumns(columns.toArray(new IcebergColumn[0])).build();
```

Comment:
it's odd to create a mockTable, why not use struct directly?

Reply:
the struct interface is tricky for some historical reasons, I think it's better to replace `IcebergTable` with `Struct`, @Clearvive  what do you think?
```java
struct(IcebergTable struct, List<T> fieldResults)
```

Reply:
we can refer to the implement of Iceberg https://github.com/apache/iceberg/blob/main/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkTypeToType.java#L67-L92

Reply:
The reason why I create a mockTable is that I can reuse the method 
`visit(IcebergTable table, ToIcebergTypeVisitor<T> visitor)`
If we do not use this method, we should implement new methods like you refer to iceberg
```
Type struct(StructType struct, List<Type> types)`
<T> T visit(StructType struct, ToIcebergTypeVisitor<T> visitor)
```
The logic of these methods are similar with existing methods. 
```
<T> T visit(IcebergTable table, ToIcebergTypeVisitor<T> visitor)
Type struct(IcebergTable struct, List<Type> types)
```

Is it reasonable for multiple methods with similar logic to exist ?
Also, it will be much appreciated if you can explain the tricky historical reasons of struct interface that you mentioned.

Reply:
Gravitino doesn't support Struct in early versions, so introduces `<T> T visit(IcebergTable table, ToIcebergTypeVisitor<T> visitor)`,  as struct is supported for now, we could do a refactor to existing code, make code more clean.

Reply:
@Clearvive could you provide more input?

Reply:
So what you suggest is that we should refactor from

Reply:
yes, this seems more clear and general.